File: app/Modules/Storefront/Http/Controllers/BeneficiariesController.php
<?php

namespace App\Modules\Storefront\Http\Controllers;

use App\Modules\Storefront\Models\Organization;
use App\Services\OrgSvcClient;
use Inertia\Inertia;

/**
 * Publiczna podstrona „O nas" (/beneficiaries) — lista węzłów organizacji
 * głównej (root organization). Odpowiednik per-organizacja:
 * UserBeneficiariesController pod /people/{handle}/wspieramy.
 */
class BeneficiariesController extends Controller
{
    public function index()
    {
        $org = Organization::rootOrganization();
        $nodes = $org ? app(OrgSvcClient::class)->listBeneficiaryNodes($org->id, activeOnly: true) : [];
        usort($nodes, fn ($a, $b) => $a['position'] <=> $b['position']);

        return Inertia::render('Storefront/Beneficiaries', [
            'nodes' => array_values(array_map(fn (array $n) => [
                'id' => $n['id'],
                'heading' => $n['heading'],
                'text_align' => $n['textAlign'],
                'image' => $n['image'] ? asset('storage/' . $n['image']) : null,
                'image_x' => $n['imageX'],
                'image_y' => $n['imageY'],
                'image_scale' => $n['imageScale'],
                'image_right' => $n['imageSide'] === 'right',
                'body_html' => $n['bodyHtml'] ?? '',
            ], $nodes)),
            'pageTitle' => 'O nas — ' . config('shop.name'),
            'pageDescription' => 'Kogo i jak wspieramy — SupportMe łączy ludzi, wartości i nowoczesne płatności.',
        ]);
    }
}

File: pay-backend/storefront-api/src/main/kotlin/com/supportme/pay/storefront/api/pub/BeneficiariesController.kt
package com.supportme.pay.storefront.api.pub

import com.supportme.pay.storefront.domain.repository.BeneficiaryNodeRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

data class BeneficiaryNodeSummary(
    val heading: String,
    val image: String?,
    val imageSide: String,
    val imageScale: Int,
    val imageX: Int,
    val imageY: Int,
    val textAlign: String,
    val bodyHtml: String?,
)

/** Odpowiednik `BeneficiariesController::index` — read-only strona „Wspieramy". */
@RestController
class BeneficiariesController(private val beneficiaryNodeRepository: BeneficiaryNodeRepository) {

    @GetMapping("/beneficiaries")
    fun index(): List<BeneficiaryNodeSummary> = beneficiaryNodeRepository.findAllByActiveTrueOrderByPositionAscIdAsc().map {
        BeneficiaryNodeSummary(it.heading, it.image, it.imageSide.dbValue, it.imageScale, it.imageX, it.imageY, it.textAlign.dbValue, it.bodyHtml)
    }
}
